### PR TITLE
[fix](memory) Fix reserve memory compatible with memory GC and logging

### PR DESCRIPTION
**Pull request by xinyiZzz**
## Proposed changes

optimize process memory info logging

### CODE REVIEW

**Comment by xinyiZzz** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

**Review by mrhhsg — DISMISSED** [review]
LGTM

File: be/src/vec/sink/writer/vtablet_writer.cpp [hunk 1]
```diff
@@ -556,8 +556,10 @@ Status VNodeChannel::add_block(vectorized::Block* block, const Payload* payload)
 
 int VNodeChannel::try_send_and_fetch_status(RuntimeState* state,
                                             std::unique_ptr<ThreadPoolToken>& thread_pool_token) {
-    DBUG_EXECUTE_IF("VNodeChannel.try_send_and_fetch_status_full_gc",
-                    { MemInfo::process_full_gc(); });
+    DBUG_EXECUTE_IF("VNodeChannel.try_send_and_fetch_status_full_gc", {
+        MemInfo::process_full_gc(
+                doris::GlobalMemoryArbitrator::process_limit_exceeded_errmsg_str());
```
Comment (HappenLee) [inline be/src/vec/sink/writer/vtablet_writer.cpp:561]: why here need pass the `doris::GlobalMemoryArbitrator::process_limit_exceeded_errmsg_str()` ? if empty() in function internal will call the func.

File: be/src/common/daemon.cpp [hunk 2]
```diff
@@ -259,26 +259,26 @@ void Daemon::memory_gc_thread() {
...
+                    doris::GlobalMemoryArbitrator::process_limit_exceeded_errmsg_str();
             memory_full_gc_sleep_time_ms = memory_gc_sleep_time_ms;
             memory_minor_gc_sleep_time_ms = memory_gc_sleep_time_ms;
-            LOG(INFO) << fmt::format(
-                    "[MemoryGC] start full GC, {}.",
-                    doris::GlobalMemoryArbitrator::process_limit_exceeded_errmsg_str());
+            LOG(INFO) << fmt::format("[MemoryGC] start full GC, {}.", mem_info);
             doris::MemTrackerLimiter::print_log_process_usage();
-            if (doris::MemInfo::process_full_gc()) {
+            if (doris::MemInfo::process_full_gc(mem_info)) {
```
Comment (HappenLee) [inline be/src/common/daemon.cpp:268]: std::move(mem_info)

File: be/src/olap/memtable_memory_limiter.cpp [hunk 3]
```diff
@@ -225,10 +225,8 @@ void MemTableMemoryLimiter::refresh_mem_tracker() {
     _log_timer.reset();
     // if not exist load task, this log should not be printed.
     if (_mem_usage != 0) {
-        LOG(INFO) << ss.str() << ", process mem: " << PerfCounters::get_vm_rss_str()
-                  << " (without allocator cache: "
-                  << PrettyPrinter::print_bytes(GlobalMemoryArbitrator::process_memory_usage())
-                  << "), load mem: " << PrettyPrinter::print_bytes(_mem_tracker->consumption())
+        LOG(INFO) << ss.str() << ", " << GlobalMemoryArbitrator::process_memory_used_details_str()
```
Comment (HappenLee) [inline be/src/olap/memtable_memory_limiter.cpp:228]: better do not use streamstring，use fmt to same work. the stringstream bad performance

File: be/src/vec/sink/writer/vtablet_writer.cpp (hunk 1 above)
Comment (xinyiZzz) [reply be/src/vec/sink/writer/vtablet_writer.cpp:561]: done

File: be/src/common/daemon.cpp (hunk 2 above)
Comment (xinyiZzz) [reply be/src/common/daemon.cpp:268]: done

**Comment by xinyiZzz** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by wangbo — APPROVED** [review]
LGTM

**Pull request merged**